### ISSUE TITLE
GPU Driven Rendering - Part One

### ISSUE THREAD

**Issue by kanerogers** [kanerogers/gambier#9]
## TODO
- [x] Read GPU Driven Rendering article in detail
- [x] Implement current rendering system with drawIndirect
  - [x] Create buffer of commands
  - [x] Record draw commands in loop
- [x] Switch to bindless textures
  - [x] Build buffer of samplers
  - [x] Only bind _once_
  - [x] Switch to fully dynamic shaders (eg. no fixed size arrays)
 - [x] Execute trivial compute shader

**Comment by kanerogers** [comment]
Computer shader:

- [x] Create compute pipeline
- [x] Bind to compute pipeline

**Comment by kanerogers** [comment]
Notes:

Things are, in general, looking pretty good. The only issue we've got at the moment is that there are too many textures to fit into memory. This is a bit of a problem. I've hacked around this in the Sponza scene by only including the base colour textures but that's a pretty terrible work around. Our model works right now by essentially loading the entire glTF file onto the goddamn GPU.

We are just about good to go with the fully dynamic descriptors, we just need to enable some extensions and device features etc. Fingers crossed!
